Bottom: 96dfad980da67407a7c93f1f4f961535edf33036
Top:    f1461e27a310bcba5bc27c10f61f02e952bdffd2
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-12-16 16:21:37 +0800

Refresh of xdp-introduce-xdp-free-func

---

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_rx.c b/drivers/net/ethernet/mellanox/mlx4/en_rx.c
index 9fef266f..9fea7e4 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_rx.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_rx.c
@@ -910,6 +910,7 @@ int mlx4_en_process_rx_cq(struct net_device *dev, struct mlx4_en_cq *cq, int bud
 			xdp.data = ring;
 			xdp.private = private;
 			xdp.free = mlx4_en_xdp_buff_free;
+			xdp.netdev = dev;
 
 			act = bpf_prog_run_xdp(xdp_prog, &xdp);
 			switch (act) {
diff --git a/include/linux/filter.h b/include/linux/filter.h
index 830d28e..13fdcd6 100644
--- a/include/linux/filter.h
+++ b/include/linux/filter.h
@@ -437,6 +437,7 @@ struct xdp_buff {
 	void *data_end;
 	void *ring;
 	void *private;
+	struct net_device *netdev;
 	void (*free) (const void *ring, const void *private);
 };
